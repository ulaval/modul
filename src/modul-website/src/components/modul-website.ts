import { ModulVue } from '@ulaval/modul-components/dist/utils/vue/vue';

export class ModulWebsite extends ModulVue {

}
