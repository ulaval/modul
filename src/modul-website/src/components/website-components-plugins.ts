import Vue, { PluginObject } from 'vue';
import { MWCard, MWCARD_NAME } from './card/card';
import { COLOR_NAME, MColor } from './color/color';
import { COMPONENT_API_NAME, MComponentApi } from './component-api/component-api';
import { DEMO_NAME, MDemo } from './demo/demo';
import { DO_NAME, MDo } from './do/do';
import { DONT_NAME, MDont } from './dont/dont';
import { MDynamicTemplate } from './dynamic-template/dynamic-template';
import { MWExpandablePanel } from './expendable-panel/expandable-panel';
import { GO_NAME, MGo } from './go/go';
import { MWHeader, MWHEADER_NAME } from './header/header';
import { HIGHLIGHT_NAME, MHighlight } from './highlight/highlight';
import { ICON_GALLERY_NAME, MIconGallery } from './icon-gallery/icon-gallery';
import { MWLogo, MWLOGO_NAME } from './logo/logo';
import { MARKDOWN_NAME, MMarkdown } from './markdown/markdown';
import { MWMegaMenu, MWMEGAMENU_NAME } from './mega-menu/mega-menu';
import { MWPanel, MWPANEL_NAME } from './panel/panel';
import { MPreview, PREVIEW_NAME } from './preview/preview';
import SliderPlugin from './slider/slider';

const WebsiteComponentsPlugin: PluginObject<any> = {
    install(v): void {
        Vue.component(COLOR_NAME, MColor);
        Vue.component(DEMO_NAME, MDemo);
        Vue.component(DO_NAME, MDo);
        Vue.component(DONT_NAME, MDont);
        Vue.component(GO_NAME, MGo);
        Vue.component(MARKDOWN_NAME, MMarkdown);
        Vue.component(PREVIEW_NAME, MPreview);
        Vue.component(MWHEADER_NAME, MWHeader);
        Vue.component(MWMEGAMENU_NAME, MWMegaMenu);
        Vue.component(MWLOGO_NAME, MWLogo);
        Vue.component(MWCARD_NAME, MWCard);
        Vue.component(MWPANEL_NAME, MWPanel);
        Vue.component(HIGHLIGHT_NAME, MHighlight);
        Vue.component(ICON_GALLERY_NAME, MIconGallery);
        Vue.component(COMPONENT_API_NAME, MComponentApi);
        Vue.component('mw-dynamic-template', MDynamicTemplate);
        Vue.component('mw-expandable-panel', MWExpandablePanel);
        Vue.component('mw-expandable-panel', MWExpandablePanel);
        Vue.use(SliderPlugin);
    }
};

export default WebsiteComponentsPlugin;
