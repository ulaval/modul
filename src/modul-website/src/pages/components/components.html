<div class="mw-components">
    <m-expandable-layout class="mw-components__body"
                         :open="showMenu"
                         panel-width="320px">
        <div class="mw-components__page">
            <m-icon-button class="mw-components__menu-button"
                           @click="toggleMenu()"
                           :icon-name="showMenu ? 'm-svg__close-clear' : 'm-standard-edit'"
                           skin="light"
                           icon-size="20px">
            </m-icon-button>

            <div class="mw-components__content">
                <mw-component-details :component="component"></mw-component-details>
            </div>
        </div>
        <template slot="panel">
            <div class="mw-components__sidemenu">

                <div class="mw-components__search">
                    <m-searchfield></m-searchfield>
                </div>
                <div class="mw-components__sidemenu-body">
                    <div class="mw-components__sidemenu-title">
                        <m-i18n k="website:components-ui"></m-i18n>
                    </div>
                    <m-accordion skin="plain"
                                 icon-position="right"
                                 icon-size="small"
                                 v-for="categoryId in componentCategoryIds"
                                 :key="categoryId"
                                 :open="urlMatchPath(categoryId)">
                        <div class="mw-components__sidemenu-sub-title"
                             slot="header">
                            <m-i18n :k="'categories:'+categoryId"></m-i18n>
                        </div>
                        <div>
                            <div class="mw-components__sidemenu-item"
                                 v-for="component in getCategoryComponents(categoryId)"
                                 :key="component.url">
                                <router-link :to="getRouterIndex(component.url)"
                                             @click.native="onSideMenuSelection()"><span>
                                        {{component.name}}
                                    </span>
                                    <span v-if="component.beta">
                                        <span class="modul-logo__beta mw-components__beta">
                                            <m-i18n k="website:beta"></m-i18n>
                                        </span>
                                    </span>
                                </router-link>
                            </div>
                        </div>
                    </m-accordion>
                </div>
            </div>
        </template>
    </m-expandable-layout>
</div>
