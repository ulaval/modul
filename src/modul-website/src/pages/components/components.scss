@import '../../styles/commons';

.mw-component-details {
    width: 100%;

    &__component-navigation {
        display: flex;
        margin-top: 54px;
        color: $m-color--interactive-light;
    }

    &__component-navigation-next,
    &__component-navigation-prev {
        line-height: 24px;
        position: relative;
        padding: 1rem;
        flex-grow: 1;
        border: 1px solid;
        text-align: center;
        text-decoration: none;
        font-weight: $m-font-weight--semi-bold;
    }

    &__component-navigation-next--icon {
        position: absolute;
        right: 16px;
    }

    &__component-navigation-prev--icon {
        position: absolute;
        left: 16px;
    }
}
