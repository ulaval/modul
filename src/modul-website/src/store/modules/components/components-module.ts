import { Getter, ModuleBase, Mutation } from '@ulaval/modul-vuex/dist/vuex-annotations';
import { PluginObject } from 'vue';
import { Store } from 'vuex';

declare module 'vue/types/vue' {
    interface Vue {
        $componentsModule: ComponentsModule;
    }
}

export interface ComponentsModuleState {
    components: { [k: string]: any | undefined };
    version: string;
}

export class ComponentsModule extends ModuleBase<ComponentsModuleState> {
    constructor(store: Store<any>, moduleName: string = 'components') {
        super(moduleName, {
            components: {},
            version: ''
        }, store);
    }

    @Mutation()
    public setVersion(version: string): void {
        this.state.version = version;
    }

    @Getter()
    public getVersion(): string {
        return this.state.version;
    }
}

export const ComponentsModulePlugin: PluginObject<any> = {
    install(v, options): void {
        if (options) {
            v.prototype.$componentsModule = new ComponentsModule(options.store);
        }

    }
};
