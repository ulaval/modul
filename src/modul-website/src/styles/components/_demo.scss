.modul-demo {
    &__preview {
        div:first-child > :first-child {
            margin-top: 0;
        }
    }
}
