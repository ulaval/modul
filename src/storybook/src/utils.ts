export const modulComponentsHierarchyRootSeparator: string = 'modul-components|';
export const modulWebsiteHierarchyRootSeparator: string = 'modul-website|';
