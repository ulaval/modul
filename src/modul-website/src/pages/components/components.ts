
import { MWExpandablePanel } from '@/components/expendable-panel/expandable-panel';
import { ModulWebsite } from '@/components/modul-website';
import { ComponentMeta } from '@/content/components.meta.loader';
import _ from 'lodash';
import Vue from 'vue';
import Component from 'vue-class-component';
import { Prop, Watch } from 'vue-property-decorator';
import { MWComponent } from './component/component-details';
import WithRender from './components.html?style=./components.scss';

@WithRender
@Component({
    components: { 'mw-component-details': MWComponent }
})
export class MWComponentsPage extends ModulWebsite {

    @Prop()
    component: ComponentMeta;

    $refs: {
        expandablePanel: MWExpandablePanel;
        prevLink: Vue;
        nextLink: Vue;
    };

    fullPath: string;

    get componentCategoryIds(): string[] {
        return _.sortBy(this.$meta.categories, name => _.deburr(this.$i18n.translate(`categories:${name}`)));
    }

    get componentsNavigationOrder(): ComponentMeta[] {
        let components: ComponentMeta[] = [];
        this.componentCategoryIds.forEach(category => {
            this.$meta.componentState[category].forEach((componentMeta: ComponentMeta) => {
                components.push(componentMeta);
            });
        });
        return components;
    }

    get currentComponentNavigationIndex(): number {
        return _.findIndex(this.componentsNavigationOrder, (c: ComponentMeta) => {
            return this.component.name === c.name;
        });
    }

    get prevComponent(): ComponentMeta {
        let index = this.currentComponentNavigationIndex;

        // if this is the first item return the last
        if (index - 1 < 0) {
            return this.componentsNavigationOrder[this.componentsNavigationOrder.length - 1];
        } else {
            return this.componentsNavigationOrder[index - 1];
        }
    }

    get nextComponent(): ComponentMeta {
        let index = this.currentComponentNavigationIndex;

        // if this is the last item return the first
        if (index + 1 >= this.componentsNavigationOrder.length - 1) {
            return this.componentsNavigationOrder[0];
        } else {
            return this.componentsNavigationOrder[index + 1];
        }
    }

    getRouterIndex(tag: string): string {
        return this.$routerIndex.for(tag);
    }

    getCategoryComponents(category): ComponentMeta[] {
        return this.$meta.componentState[category];
    }

    urlMatchPath(categoryId: string) {
        let path: string = this.$i18n.translate(`categories:${categoryId}-route`);
        return this.fullPath.includes(`/${path}`);
    }

    onSideMenuSelection() {
        this.$refs.expandablePanel.onSideMenuSelection();
    }

    onPrevLinkClick() {
        (this.$refs.prevLink.$el as HTMLElement).blur();
    }

    onNextLinkClick() {
        (this.$refs.nextLink.$el as HTMLElement).blur();
    }

    @Watch('$route', { immediate: true })
    private routeChanged(): void {
        this.fullPath = this.$route.fullPath;
    }
}
