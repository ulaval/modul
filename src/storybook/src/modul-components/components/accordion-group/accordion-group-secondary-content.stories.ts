import { ACCORDION_GROUP_NAME } from '@ulaval/modul-components/dist/components/component-names';
import { modulComponentsHierarchyRootSeparator } from './../../../utils';
export default {
    title: `${modulComponentsHierarchyRootSeparator}${ACCORDION_GROUP_NAME}/secondary-content`,
    parameters: { fileName: __filename }
};

export const defaultStory = () => ({
    template: `
    <m-accordion-group :openedIds="[1,2,3]">
        <m-accordion :id="1"><h3 slot="header">An Accordion Title</h3>Some Accordion Content</m-accordion>
        <m-accordion :id="2"><h3 slot="header">An Accordion Title</h3>Some Accordion Content</m-accordion>
        <m-accordion :id="3"><h3 slot="header">An Accordion Title</h3>Some Accordion Content</m-accordion>
    </m-accordion-group>`
});
defaultStory.story = { name: 'default' };

export const toggleLinkLeft = () => ({
    template: `
    <m-accordion-group>
        <h2 slot="title">An Accordion Group Title</h2>
        <h3 slot="secondary-content">An Accordion Group Title</h3>
        <m-accordion><h3 slot="header">An Accordion Title</h3>Some Accordion Content</m-accordion>
        <m-accordion><h3 slot="header">An Accordion Title</h3>Some Accordion Content</m-accordion>
        <m-accordion><h3 slot="header">An Accordion Title</h3>Some Accordion Content</m-accordion>
    </m-accordion-group>`
});
toggleLinkLeft.story = { name: 'toggle-link-left' };

export const link = () => ({
    template: `
    <m-accordion-group :toggle-link-left="true">
        <h2 slot="title">An Accordion Group Title</h2>
        <m-link icon-name="m-svg__add-circle-filled" icon-size="22px" mode="button" slot="secondary-content">Add content</m-link>
        <m-accordion><h3 slot="header">An Accordion Title</h3>Some Accordion Content</m-accordion>
        <m-accordion><h3 slot="header">An Accordion Title</h3>Some Accordion Content</m-accordion>
        <m-accordion><h3 slot="header">An Accordion Title</h3>Some Accordion Content</m-accordion>
    </m-accordion-group>`
});
link.story = { name: 'm-link' };

export const noTitle = () => ({
    template: `<m-accordion-group>
                    <h3 slot="secondary-content">An Accordion Group Title</h3>
                    <m-accordion><h3 slot="header">An Accordion Title</h3>Some Accordion Content</m-accordion>
                    <m-accordion><h3 slot="header">An Accordion Title</h3>Some Accordion Content</m-accordion>
                    <m-accordion><h3 slot="header">An Accordion Title</h3>Some Accordion Content</m-accordion>
</m-accordion-group>`
});
noTitle.story = { name: 'no-title' };

export const noToggleLinkLeft = () => ({
    template: `
    <m-accordion-group :toggle-link-left="true">
        <h3 slot="secondary-content">An Accordion Group Title</h3>
        <m-accordion><h3 slot="header">An Accordion Title</h3>Some Accordion Content</m-accordion>
        <m-accordion><h3 slot="header">An Accordion Title</h3>Some Accordion Content</m-accordion>
        <m-accordion><h3 slot="header">An Accordion Title</h3>Some Accordion Content</m-accordion>
    </m-accordion-group>`
});
noToggleLinkLeft.story = { name: 'no-title-toggle-link-left' };


