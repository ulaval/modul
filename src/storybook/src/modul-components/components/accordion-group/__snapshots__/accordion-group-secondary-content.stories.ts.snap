// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots modul-components|m-accordion-group/secondary-content default 1`] = `
<article
  class="m-accordion-group"
>
  <div
    class="m-accordion-group__header m--no-title"
  >
    <!---->
     
    <div
      class="m-accordion-group__secondary-content"
    >
       
      <div
        class="m-accordion-group__toggle-link"
      >
        <a
          class="m-link m--is-unvisited m--is-button m--has-content"
          href="#"
          role="button"
          tabindex="0"
        >
          <!---->
           
          <span
            class="m-link__text"
          >
            <span
              class="m-i18n"
            >
              Tout ouvrir
            </span>
          </span>
           
          <!---->
        </a>
           |   
        <a
          class="m-link m--is-unvisited m--is-button m--has-content"
          href="#"
          role="button"
          tabindex="0"
        >
          <!---->
           
          <span
            class="m-link__text"
          >
            <span
              class="m-i18n"
            >
              Tout fermer
            </span>
          </span>
           
          <!---->
        </a>
      </div>
    </div>
  </div>
   
  <div
    class="m-accordion-group__body"
  >
    <article
      class="m-accordion m--is-default m--is-open m--has-icon-left m--has-icon-large"
    >
      <div
        aria-controls="1-body-wrap"
        aria-expanded="true"
        class="m-accordion__header m--has-padding"
        id="1"
        role="tab"
        tabindex="0"
      >
        <div
          class="m-accordion__header-content"
        >
          <h3>
            An Accordion Title
          </h3>
           
          <!---->
           
          <span
            class="m-i18n m-accordion__hidden"
          >
            Cliquer pour cacher le contenu
          </span>
        </div>
         
        <span
          aria-hidden="true"
          class="m-plus m-accordion__header-icon m--is-skin-default m--is-open m--is-large m--is-left"
          title="Cliquer pour cacher le contenu"
        />
      </div>
       
      <div
        aria-labelledby="1"
        class="m-accordion__body-wrap"
        id="1-body-wrap"
        role="tabpanel"
      >
        <div
          class="m-accordion__body m--has-padding"
        >
          Some Accordion Content
        </div>
      </div>
    </article>
     
    <article
      class="m-accordion m--is-default m--is-open m--has-icon-left m--has-icon-large"
    >
      <div
        aria-controls="2-body-wrap"
        aria-expanded="true"
        class="m-accordion__header m--has-padding"
        id="2"
        role="tab"
        tabindex="0"
      >
        <div
          class="m-accordion__header-content"
        >
          <h3>
            An Accordion Title
          </h3>
           
          <!---->
           
          <span
            class="m-i18n m-accordion__hidden"
          >
            Cliquer pour cacher le contenu
          </span>
        </div>
         
        <span
          aria-hidden="true"
          class="m-plus m-accordion__header-icon m--is-skin-default m--is-open m--is-large m--is-left"
          title="Cliquer pour cacher le contenu"
        />
      </div>
       
      <div
        aria-labelledby="2"
        class="m-accordion__body-wrap"
        id="2-body-wrap"
        role="tabpanel"
      >
        <div
          class="m-accordion__body m--has-padding"
        >
          Some Accordion Content
        </div>
      </div>
    </article>
     
    <article
      class="m-accordion m--is-default m--is-open m--has-icon-left m--has-icon-large"
    >
      <div
        aria-controls="3-body-wrap"
        aria-expanded="true"
        class="m-accordion__header m--has-padding"
        id="3"
        role="tab"
        tabindex="0"
      >
        <div
          class="m-accordion__header-content"
        >
          <h3>
            An Accordion Title
          </h3>
           
          <!---->
           
          <span
            class="m-i18n m-accordion__hidden"
          >
            Cliquer pour cacher le contenu
          </span>
        </div>
         
        <span
          aria-hidden="true"
          class="m-plus m-accordion__header-icon m--is-skin-default m--is-open m--is-large m--is-left"
          title="Cliquer pour cacher le contenu"
        />
      </div>
       
      <div
        aria-labelledby="3"
        class="m-accordion__body-wrap"
        id="3-body-wrap"
        role="tabpanel"
      >
        <div
          class="m-accordion__body m--has-padding"
        >
          Some Accordion Content
        </div>
      </div>
    </article>
  </div>
</article>
`;

exports[`Storyshots modul-components|m-accordion-group/secondary-content m-link 1`] = `
<article
  class="m-accordion-group"
>
  <div
    class="m-accordion-group__header m--has-secondary-content m--is-toggle-link-left"
  >
    <div
      class="m-accordion-group__header-title"
    >
      <h2>
        An Accordion Group Title
      </h2>
    </div>
     
    <div
      class="m-accordion-group__secondary-content"
    >
      <a
        class="m-link m--is-unvisited m--is-button m--has-left-icon m--has-content"
        href="#"
        role="button"
        tabindex="0"
      >
        <svg
          aria-hidden="true"
          class="m-icon m-link__icon"
          height="22px"
          width="22px"
        >
          <!---->
           
          <use
            aria-hidden="true"
            class=""
          />
        </svg>
         
        <span
          class="m-link__text"
        >
          Add content
        </span>
         
        <!---->
      </a>
       
      <div
        class="m-accordion-group__toggle-link"
      >
        <a
          class="m-link m--is-unvisited m--is-button m--has-content"
          href="#"
          role="button"
          tabindex="0"
        >
          <!---->
           
          <span
            class="m-link__text"
          >
            <span
              class="m-i18n"
            >
              Tout ouvrir
            </span>
          </span>
           
          <!---->
        </a>
           |   
        <a
          class="m-link m--is-unvisited m--is-button m--has-content"
          href="#"
          role="button"
          tabindex="0"
        >
          <!---->
           
          <span
            class="m-link__text"
          >
            <span
              class="m-i18n"
            >
              Tout fermer
            </span>
          </span>
           
          <!---->
        </a>
      </div>
    </div>
  </div>
   
  <div
    class="m-accordion-group__body"
  >
      
    <article
      class="m-accordion m--is-default m--is-closed m--has-icon-left m--has-icon-large"
    >
      <div
        aria-controls="fakeId-body-wrap"
        class="m-accordion__header m--has-padding"
        id="fakeId"
        role="tab"
        tabindex="0"
      >
        <div
          class="m-accordion__header-content"
        >
          <h3>
            An Accordion Title
          </h3>
           
          <span
            class="m-i18n m-accordion__hidden"
          >
            Cliquer pour afficher le contenu
          </span>
           
          <!---->
        </div>
         
        <span
          aria-hidden="true"
          class="m-plus m-accordion__header-icon m--is-skin-default m--is-large m--is-left"
          title="Cliquer pour afficher le contenu"
        />
      </div>
       
      <!---->
    </article>
     
    <article
      class="m-accordion m--is-default m--is-closed m--has-icon-left m--has-icon-large"
    >
      <div
        aria-controls="fakeId-body-wrap"
        class="m-accordion__header m--has-padding"
        id="fakeId"
        role="tab"
        tabindex="0"
      >
        <div
          class="m-accordion__header-content"
        >
          <h3>
            An Accordion Title
          </h3>
           
          <span
            class="m-i18n m-accordion__hidden"
          >
            Cliquer pour afficher le contenu
          </span>
           
          <!---->
        </div>
         
        <span
          aria-hidden="true"
          class="m-plus m-accordion__header-icon m--is-skin-default m--is-large m--is-left"
          title="Cliquer pour afficher le contenu"
        />
      </div>
       
      <!---->
    </article>
     
    <article
      class="m-accordion m--is-default m--is-closed m--has-icon-left m--has-icon-large"
    >
      <div
        aria-controls="fakeId-body-wrap"
        class="m-accordion__header m--has-padding"
        id="fakeId"
        role="tab"
        tabindex="0"
      >
        <div
          class="m-accordion__header-content"
        >
          <h3>
            An Accordion Title
          </h3>
           
          <span
            class="m-i18n m-accordion__hidden"
          >
            Cliquer pour afficher le contenu
          </span>
           
          <!---->
        </div>
         
        <span
          aria-hidden="true"
          class="m-plus m-accordion__header-icon m--is-skin-default m--is-large m--is-left"
          title="Cliquer pour afficher le contenu"
        />
      </div>
       
      <!---->
    </article>
  </div>
</article>
`;

exports[`Storyshots modul-components|m-accordion-group/secondary-content no-title 1`] = `
<article
  class="m-accordion-group"
>
  <div
    class="m-accordion-group__header m--no-title m--has-secondary-content"
  >
    <!---->
     
    <div
      class="m-accordion-group__secondary-content"
    >
      <h3>
        An Accordion Group Title
      </h3>
       
      <div
        class="m-accordion-group__toggle-link"
      >
        <a
          class="m-link m--is-unvisited m--is-button m--has-content"
          href="#"
          role="button"
          tabindex="0"
        >
          <!---->
           
          <span
            class="m-link__text"
          >
            <span
              class="m-i18n"
            >
              Tout ouvrir
            </span>
          </span>
           
          <!---->
        </a>
           |   
        <a
          class="m-link m--is-unvisited m--is-button m--has-content"
          href="#"
          role="button"
          tabindex="0"
        >
          <!---->
           
          <span
            class="m-link__text"
          >
            <span
              class="m-i18n"
            >
              Tout fermer
            </span>
          </span>
           
          <!---->
        </a>
      </div>
    </div>
  </div>
   
  <div
    class="m-accordion-group__body"
  >
     
    <article
      class="m-accordion m--is-default m--is-closed m--has-icon-left m--has-icon-large"
    >
      <div
        aria-controls="fakeId-body-wrap"
        class="m-accordion__header m--has-padding"
        id="fakeId"
        role="tab"
        tabindex="0"
      >
        <div
          class="m-accordion__header-content"
        >
          <h3>
            An Accordion Title
          </h3>
           
          <span
            class="m-i18n m-accordion__hidden"
          >
            Cliquer pour afficher le contenu
          </span>
           
          <!---->
        </div>
         
        <span
          aria-hidden="true"
          class="m-plus m-accordion__header-icon m--is-skin-default m--is-large m--is-left"
          title="Cliquer pour afficher le contenu"
        />
      </div>
       
      <!---->
    </article>
     
    <article
      class="m-accordion m--is-default m--is-closed m--has-icon-left m--has-icon-large"
    >
      <div
        aria-controls="fakeId-body-wrap"
        class="m-accordion__header m--has-padding"
        id="fakeId"
        role="tab"
        tabindex="0"
      >
        <div
          class="m-accordion__header-content"
        >
          <h3>
            An Accordion Title
          </h3>
           
          <span
            class="m-i18n m-accordion__hidden"
          >
            Cliquer pour afficher le contenu
          </span>
           
          <!---->
        </div>
         
        <span
          aria-hidden="true"
          class="m-plus m-accordion__header-icon m--is-skin-default m--is-large m--is-left"
          title="Cliquer pour afficher le contenu"
        />
      </div>
       
      <!---->
    </article>
     
    <article
      class="m-accordion m--is-default m--is-closed m--has-icon-left m--has-icon-large"
    >
      <div
        aria-controls="fakeId-body-wrap"
        class="m-accordion__header m--has-padding"
        id="fakeId"
        role="tab"
        tabindex="0"
      >
        <div
          class="m-accordion__header-content"
        >
          <h3>
            An Accordion Title
          </h3>
           
          <span
            class="m-i18n m-accordion__hidden"
          >
            Cliquer pour afficher le contenu
          </span>
           
          <!---->
        </div>
         
        <span
          aria-hidden="true"
          class="m-plus m-accordion__header-icon m--is-skin-default m--is-large m--is-left"
          title="Cliquer pour afficher le contenu"
        />
      </div>
       
      <!---->
    </article>
  </div>
</article>
`;

exports[`Storyshots modul-components|m-accordion-group/secondary-content no-title-toggle-link-left 1`] = `
<article
  class="m-accordion-group"
>
  <div
    class="m-accordion-group__header m--no-title m--has-secondary-content m--is-toggle-link-left"
  >
    <!---->
     
    <div
      class="m-accordion-group__secondary-content"
    >
      <h3>
        An Accordion Group Title
      </h3>
       
      <div
        class="m-accordion-group__toggle-link"
      >
        <a
          class="m-link m--is-unvisited m--is-button m--has-content"
          href="#"
          role="button"
          tabindex="0"
        >
          <!---->
           
          <span
            class="m-link__text"
          >
            <span
              class="m-i18n"
            >
              Tout ouvrir
            </span>
          </span>
           
          <!---->
        </a>
           |   
        <a
          class="m-link m--is-unvisited m--is-button m--has-content"
          href="#"
          role="button"
          tabindex="0"
        >
          <!---->
           
          <span
            class="m-link__text"
          >
            <span
              class="m-i18n"
            >
              Tout fermer
            </span>
          </span>
           
          <!---->
        </a>
      </div>
    </div>
  </div>
   
  <div
    class="m-accordion-group__body"
  >
     
    <article
      class="m-accordion m--is-default m--is-closed m--has-icon-left m--has-icon-large"
    >
      <div
        aria-controls="fakeId-body-wrap"
        class="m-accordion__header m--has-padding"
        id="fakeId"
        role="tab"
        tabindex="0"
      >
        <div
          class="m-accordion__header-content"
        >
          <h3>
            An Accordion Title
          </h3>
           
          <span
            class="m-i18n m-accordion__hidden"
          >
            Cliquer pour afficher le contenu
          </span>
           
          <!---->
        </div>
         
        <span
          aria-hidden="true"
          class="m-plus m-accordion__header-icon m--is-skin-default m--is-large m--is-left"
          title="Cliquer pour afficher le contenu"
        />
      </div>
       
      <!---->
    </article>
     
    <article
      class="m-accordion m--is-default m--is-closed m--has-icon-left m--has-icon-large"
    >
      <div
        aria-controls="fakeId-body-wrap"
        class="m-accordion__header m--has-padding"
        id="fakeId"
        role="tab"
        tabindex="0"
      >
        <div
          class="m-accordion__header-content"
        >
          <h3>
            An Accordion Title
          </h3>
           
          <span
            class="m-i18n m-accordion__hidden"
          >
            Cliquer pour afficher le contenu
          </span>
           
          <!---->
        </div>
         
        <span
          aria-hidden="true"
          class="m-plus m-accordion__header-icon m--is-skin-default m--is-large m--is-left"
          title="Cliquer pour afficher le contenu"
        />
      </div>
       
      <!---->
    </article>
     
    <article
      class="m-accordion m--is-default m--is-closed m--has-icon-left m--has-icon-large"
    >
      <div
        aria-controls="fakeId-body-wrap"
        class="m-accordion__header m--has-padding"
        id="fakeId"
        role="tab"
        tabindex="0"
      >
        <div
          class="m-accordion__header-content"
        >
          <h3>
            An Accordion Title
          </h3>
           
          <span
            class="m-i18n m-accordion__hidden"
          >
            Cliquer pour afficher le contenu
          </span>
           
          <!---->
        </div>
         
        <span
          aria-hidden="true"
          class="m-plus m-accordion__header-icon m--is-skin-default m--is-large m--is-left"
          title="Cliquer pour afficher le contenu"
        />
      </div>
       
      <!---->
    </article>
  </div>
</article>
`;

exports[`Storyshots modul-components|m-accordion-group/secondary-content toggle-link-left 1`] = `
<article
  class="m-accordion-group"
>
  <div
    class="m-accordion-group__header m--has-secondary-content"
  >
    <div
      class="m-accordion-group__header-title"
    >
      <h2>
        An Accordion Group Title
      </h2>
    </div>
     
    <div
      class="m-accordion-group__secondary-content"
    >
      <h3>
        An Accordion Group Title
      </h3>
       
      <div
        class="m-accordion-group__toggle-link"
      >
        <a
          class="m-link m--is-unvisited m--is-button m--has-content"
          href="#"
          role="button"
          tabindex="0"
        >
          <!---->
           
          <span
            class="m-link__text"
          >
            <span
              class="m-i18n"
            >
              Tout ouvrir
            </span>
          </span>
           
          <!---->
        </a>
           |   
        <a
          class="m-link m--is-unvisited m--is-button m--has-content"
          href="#"
          role="button"
          tabindex="0"
        >
          <!---->
           
          <span
            class="m-link__text"
          >
            <span
              class="m-i18n"
            >
              Tout fermer
            </span>
          </span>
           
          <!---->
        </a>
      </div>
    </div>
  </div>
   
  <div
    class="m-accordion-group__body"
  >
      
    <article
      class="m-accordion m--is-default m--is-closed m--has-icon-left m--has-icon-large"
    >
      <div
        aria-controls="fakeId-body-wrap"
        class="m-accordion__header m--has-padding"
        id="fakeId"
        role="tab"
        tabindex="0"
      >
        <div
          class="m-accordion__header-content"
        >
          <h3>
            An Accordion Title
          </h3>
           
          <span
            class="m-i18n m-accordion__hidden"
          >
            Cliquer pour afficher le contenu
          </span>
           
          <!---->
        </div>
         
        <span
          aria-hidden="true"
          class="m-plus m-accordion__header-icon m--is-skin-default m--is-large m--is-left"
          title="Cliquer pour afficher le contenu"
        />
      </div>
       
      <!---->
    </article>
     
    <article
      class="m-accordion m--is-default m--is-closed m--has-icon-left m--has-icon-large"
    >
      <div
        aria-controls="fakeId-body-wrap"
        class="m-accordion__header m--has-padding"
        id="fakeId"
        role="tab"
        tabindex="0"
      >
        <div
          class="m-accordion__header-content"
        >
          <h3>
            An Accordion Title
          </h3>
           
          <span
            class="m-i18n m-accordion__hidden"
          >
            Cliquer pour afficher le contenu
          </span>
           
          <!---->
        </div>
         
        <span
          aria-hidden="true"
          class="m-plus m-accordion__header-icon m--is-skin-default m--is-large m--is-left"
          title="Cliquer pour afficher le contenu"
        />
      </div>
       
      <!---->
    </article>
     
    <article
      class="m-accordion m--is-default m--is-closed m--has-icon-left m--has-icon-large"
    >
      <div
        aria-controls="fakeId-body-wrap"
        class="m-accordion__header m--has-padding"
        id="fakeId"
        role="tab"
        tabindex="0"
      >
        <div
          class="m-accordion__header-content"
        >
          <h3>
            An Accordion Title
          </h3>
           
          <span
            class="m-i18n m-accordion__hidden"
          >
            Cliquer pour afficher le contenu
          </span>
           
          <!---->
        </div>
         
        <span
          aria-hidden="true"
          class="m-plus m-accordion__header-icon m--is-skin-default m--is-large m--is-left"
          title="Cliquer pour afficher le contenu"
        />
      </div>
       
      <!---->
    </article>
  </div>
</article>
`;
